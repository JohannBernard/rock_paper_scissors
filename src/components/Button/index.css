.my_coach_button {
  line-height: 32px;
  font-size: 32px;
  padding: 16px;
  border-radius: 16px;
  display: inline-block;
  white-space: nowrap;
  border: none;
  font-family: 'Poppins', sans-serif;
  cursor: pointer;
  color:#FFFFFF;
  background-color: #424242;
  transition: 0.2s ease-out;
  min-width: 164px;
  margin: 8px 16px;
}

.my_coach_button:hover {
  background-color: #000000;
  color:#FFEB01;
}

.my_coach_button_icon{
  min-width: auto;
  margin: 0;
}

/* Responsive */
@media screen and (max-width: 800px) {
  .my_coach_button {
    font-size: 20px;
    margin: 4px;
  }

  .my_coach_button {
    padding: 8px;
  }
}

