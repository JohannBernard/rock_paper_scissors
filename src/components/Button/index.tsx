import './index.css'

interface IButtonProps {
    label: string
    onClickCustom?: React.MouseEventHandler<HTMLButtonElement>
    icon?: boolean
    title?: string
}

function Button(props: IButtonProps) {
    return (
        <button
            onClick={props.onClickCustom}
            className={
                'my_coach_button' + (props.icon ? ' my_coach_button_icon' : '')
            }
            data-testid="testButton"
            title={props.title}
        >
            {props.label}
        </button>
    )
}

export default Button
