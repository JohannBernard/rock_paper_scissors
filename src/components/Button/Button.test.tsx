import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Button from './index'

afterEach(() => {
    cleanup()
})

test('should render button component', () => {
    render(<Button label="Test" />)
    const buttonElement = screen.getByTestId('testButton')
    expect(buttonElement).toBeInTheDocument()
    expect(buttonElement).toHaveTextContent('Test')
})

test('matches button snapshot', () => {
    const tree = renderer.create(<Button label="Test" />).toJSON()
    expect(tree).toMatchSnapshot()
})
