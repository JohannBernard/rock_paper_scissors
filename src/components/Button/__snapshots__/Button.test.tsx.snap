// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches button snapshot 1`] = `
<button
  className="my_coach_button"
  data-testid="testButton"
>
  Test
</button>
`;
